sparkmagic/sparkmagic/livyclientlib/command.py

```diff
@@ -45,6 +48,17 @@ def execute(self, session):
 
     def _get_statement_output(self, session, statement_id):
         retries = 1
+        progress = FloatProgress(value=0.0,
+                                     min=0,
+                                     max=1.0,
+                                     step=0.01,
+                                     description='Progress:',
+                                     bar_style='info',
+                                     orientation='horizontal',
+                                     layout=Layout(width='50%', height='25px')
+                                     )
+        display(progress)
```

Comment:
add an `if sys.stdout.isatty(): ` 
so progress bar will be displayed only on interactive terminals.
this should make the Travis test happy too.

Reply:
I ... am not sure that's the solution? Maybe it is, but that seems unexpected to me.

Reply:
We don't need a progress bar on non-interactive terminals (nobody can see that progressbar anyway).
So this would both suppress progress bar on such non-tty terminals,
and make Travis happy. 
I *think* Travis runs tests in non-interactive terminals too.
Thanks.

Reply:
But half the time the `display()` isn't writing to a terminal, it's writing to HTML. And when I checked, `sys.stdout.isatty()` is False for me in a Jupyter Notebook. Which means the progress bar will _never_ be displayed.

Reply:
Hmm.. that's not what I expected . 
Thanks for checking

Reply:
i might have a clue.
In the log it says
``` 
  File "/home/travis/virtualenv/python3.6.7/lib/python3.6/site-packages/ipywidgets/widgets/widget.py", line 727, in _ipython_display_
    display(data, raw=True)
```
raw=true might be the problem.

because 
```
from IPython.display import display
from ipywidgets.widgets import FloatProgress, Layout
progress = FloatProgress(value=0.0)
#display(progress, raw=True) # crashes
display(progress, raw=False) # works
```

```
help(display)
...
    raw : bool, optional
        Are the objects to be displayed already mimetype-keyed dicts of raw display data,
        or Python objects that need to be formatted before display? [default: False]
...
```

but i do not know travis enough to know what exactly generates the display() call with the raw=True parameter in the function https://github.com/jupyter-incubator/sparkmagic/blob/b7f9be6c13653d3a7402d21b8ee4aff73d0ac66e/sparkmagic/sparkmagic/tests/test_livysession.py#L437

I only found this https://github.com/jupyter-incubator/sparkmagic/blob/b7f9be6c13653d3a7402d21b8ee4aff73d0ac66e/sparkmagic/sparkmagic/tests/test_livysession.py#L83 which leads to magicmock